File: src/main/java/org/lanternpowered/server/text/format/LanternTextColor.java
package org.lanternpowered.server.text.format;

import static com.google.common.base.Preconditions.checkNotNull;

import java.awt.Color;

import org.lanternpowered.server.catalog.LanternCatalogType;
import org.spongepowered.api.text.format.TextColor;

public class LanternTextColor extends LanternCatalogType implements TextColor {

    private final Color color;

    public LanternTextColor(String name, Color color) {
        super(name, name.toLowerCase());
        this.color = checkNotNull(color, "color");
    }

    @Override
    public Color getColor() {
        return this.color;
    }
}

File: src/main/kotlin/org/lanternpowered/server/text/format/LanternTextColor.kt
/*
 * Lantern
 *
 * Copyright (c) LanternPowered <https://www.lanternpowered.org>
 * Copyright (c) SpongePowered <https://www.spongepowered.org>
 * Copyright (c) contributors
 *
 * This work is licensed under the terms of the MIT License (MIT). For
 * a copy, see 'LICENSE.txt' or <https://opensource.org/licenses/MIT>.
 */
package org.lanternpowered.server.text.format

import org.lanternpowered.api.catalog.CatalogKey
import org.lanternpowered.server.catalog.DefaultCatalogType
import org.spongepowered.api.text.format.TextColor
import org.spongepowered.api.util.Color

class LanternTextColor(key: CatalogKey, private val color: Color) : DefaultCatalogType(key), TextColor {
    override fun getColor() = this.color
    override fun toStringHelper() = super.toStringHelper()
            .add("color", this.color)
}
